src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -824,20 +824,20 @@ public function testTheEmptyStringIsAValidMappingKey()
     /**
      * @dataProvider getNotPhpCompatibleMappingKeyData
      */
-    public function testImplicitStringCastingOfMappingKeysIsDeprecated($yaml, $expected)
+    public function testImplicitStringCastingOfMappingKeysThrowsException(string $yaml)
     {
         $this->expectException(ParseException::class);
         $this->expectExceptionMessage('Implicit casting of incompatible mapping keys to strings is not supported. Quote your evaluable mapping keys instead');
-        $this->assertSame($expected, Inline::parse($yaml));
+        Inline::parse($yaml);
     }
 
-    public static function getNotPhpCompatibleMappingKeyData()
+    public static function getNotPhpCompatibleMappingKeyData(): array
     {
         return [
-            'boolean-true' => ['{true: "foo"}', ['true' => 'foo']],
-            'boolean-false' => ['{false: "foo"}', ['false' => 'foo']],
-            'null' => ['{null: "foo"}', ['null' => 'foo']],
-            'float' => ['{0.25: "foo"}', ['0.25' => 'foo']],
+            'boolean-true' => ['{true: "foo"}'],
+            'boolean-false' => ['{false: "foo"}'],
+            'null' => ['{null: "foo"}'],
+            'float' => ['{0.25: "foo"}'],
```

Comment:
not sure this rework is correct, but I think this was a deprecation before, which is now an exception, so no need to assert and no need to provide an excepted value from the dataprovider